airflow/providers_manager.py

```diff
@@ -138,7 +139,57 @@ def _check_builtin_provider_prefix(provider_package: str, class_name: str) -> bo
     return True
 
 
-def _sanity_check(provider_package: str, class_name: str) -> Optional[Type[BaseHook]]:
+@dataclass
+class ProviderInfo:
+    """Provider information"""
+
+    version: str
+    provider_info: Dict
+    package_or_source: str
```

Comment:
```suggestion
    package_or_source: Union[Literal["package"], Literal["source"]]
```

Reply:
good idear

Reply:
Yep

Reply:
"poor man's enum" - but in this case it does the job beatifully :)